src/ipc/ipc4/dai.c

```diff
@@ -79,8 +79,8 @@ int dai_config_dma_channel(struct dai_data *dd, struct comp_dev *dev, const void
 	case SOF_DAI_INTEL_DMIC:
 		channel = 0;
 #if defined(CONFIG_ACE_VERSION_2_0)
-		if (dai->host_dma_config->pre_allocated_by_host)
-			channel = dai->host_dma_config->dma_channel_id;
+		if (dai->host_dma_config[0]->pre_allocated_by_host)
+			channel = dai->host_dma_config[0]->dma_channel_id;
```

Comment:
I would break this patch in two, first the addition of this array for the host_dma_config....